nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -6,17 +6,13 @@ let
   cfg = config.services.nextcloud;
   fpm = config.services.phpfpm.pools.nextcloud;
 
-  phpPackage =
-    let
-      base = pkgs.php74;
-    in
-      base.buildEnv {
-        extensions = { enabled, all }: with all;
-          enabled ++ [
-            apcu redis memcached imagick
-          ];
-        extraConfig = phpOptionsStr;
-      };
+  phpPackage = pkgs.php74.buildEnv {
+    extensions = { enabled, all }:
+      enabled
+      ++ (with all; [ apcu redis memcached imagick ]) # Necessary for vanilla nextcloud
```

Comment:
Are `apcu`, `redis`, and `memcached` if `caching.apcu`, `caching.redis`, and `caching.memcached` are set to `false`?

Reply:
Hmm. Good question. Maybe not. With the new change, we could definitely make these extensions dependent on the options, and it's a good idea if it works.

It seems that the memcached option is actually never used for anything. Is that right?

Reply:
My impression is that in this commit, the extension setting was changed from optionally adding the three caching extensions to always adding them: https://github.com/NixOS/nixpkgs/commit/ed20aae86c4538ce2d2bf35515bb660de9a02fb9#diff-80ddb2d38f0f6e01125ccb8683381d7be20be1df3dce01e5c200dd74f5173623

So probably the right way forward now is to make them optional again, since this is easy now.

Reply:
No. See `nixos/tests/nextcloud/with-mysql-and-memcached.nix`.

> Are apcu, redis, and memcached needed if caching.apcu, caching.redis, and caching.memcached are set to false?

I don't know. And I'd be especially cautious with `apcu` as this is something that's something where people often assume that it's "just available".

Is it really worth making this module even more complex to save the closure footprint of a single php extension though?

Reply:
In that case I would recommend removing those options if they do nothing.

Reply:
> No. See nixos/tests/nextcloud/with-mysql-and-memcached.nix.

What do you mean? That test manually sets up memcached.

> Is it really worth making this module even more complex to save the closure footprint of a single php extension though?

I don't know. Does the PHP extension for redis pull in redis-the-service/package as an additional dependency? Then it would be worth making it configurable. If it's just a few small PHP files, then I agree with @aanderse that the options should be removed, and the documentation adjusted that these extensions can always be assumed to be available.

Reply:
@Ma27 how shall we go forward?